pinot-core/src/test/java/org/apache/pinot/queries/FSTBasedRegexpLikeQueriesTest.java

```diff
@@ -0,0 +1,249 @@
+            rows.add(row);
+        }
+        return rows;
+    }
+
+    private void buildSegment()
+            throws Exception {
+        List<GenericRow> rows = createTestData(NUM_ROWS);
+        List<FieldConfig> fieldConfigs = new ArrayList<>();
+        fieldConfigs.add(new FieldConfig(
+                DOMAIN_NAMES_COL,
+                FieldConfig.EncodingType.DICTIONARY,
+                FieldConfig.IndexType.FST, null));
+        fieldConfigs.add(new FieldConfig(
+                URL_COL,
+                FieldConfig.EncodingType.DICTIONARY,
+                FieldConfig.IndexType.FST, null));
+
+        TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME)
+                .setInvertedIndexColumns(Arrays.asList(DOMAIN_NAMES_COL))
+                .setFieldConfigList(fieldConfigs).build();
+        Schema schema = new Schema.SchemaBuilder().setSchemaName(TABLE_NAME)
+                .addSingleValueDimension(DOMAIN_NAMES_COL, FieldSpec.DataType.STRING)
+                .addSingleValueDimension(URL_COL, FieldSpec.DataType.STRING)
+                .addMetric(INT_COL_NAME, FieldSpec.DataType.INT).build();
+        SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, schema);
+        config.setOutDir(INDEX_DIR.getPath());
+        config.setTableName(TABLE_NAME);
+        config.setSegmentName(SEGMENT_NAME);
+
+        SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+        try (RecordReader recordReader = new GenericRowRecordReader(rows)) {
+            driver.init(config, recordReader);
+            driver.build();
+        }
+    }
+
+    private void testSelectionResults(String query, int expectedResultSize)
+            throws Exception {
+        Operator<IntermediateResultsBlock> operator = getOperatorForPqlQuery(query);
+        IntermediateResultsBlock operatorResult = operator.nextBlock();
+        List<Object[]> resultset = (List<Object[]>) operatorResult.getSelectionResult();
+        Assert.assertNotNull(resultset);
+        Assert.assertEquals(resultset.size(), expectedResultSize);
+    }
+
+    private AggregationGroupByResult getGroupByResults(String query)
+            throws Exception {
+        AggregationGroupByOperator operator = getOperatorForPqlQuery(query);
+        IntermediateResultsBlock resultsBlock = operator.nextBlock();
+        return resultsBlock.getAggregationGroupByResult();
+    }
+
+    private void matchGroupResult(AggregationGroupByResult result, String key, long count) {
+        Iterator<GroupKeyGenerator.GroupKey> groupKeyIterator = result.getGroupKeyIterator();
+        while (groupKeyIterator.hasNext()) {
+            GroupKeyGenerator.GroupKey groupKey = groupKeyIterator.next();
+            Assert.assertEquals(((Number)result.getResultForKey(groupKey, 0)).longValue(), count);
+        }
+    }
+
+    @Test
+    public void testFSTBasedRegexpLike() throws Exception {
```

Comment:
I suggest adding tests for both aggregation and selection queries. 
Also, test both segment level and inter segment execution. I initially didn't add text index tests for inter segment and there were few multi-threading related bugs in how I was using lucene parser.
Test both PQL and SQL. You may want to see example tests in TextSearchQueriesTest.java for sample tests.

Reply:
done, let me know if I should add more tests.